src/plots/cartesian/set_convert.js

```diff
@@ -182,8 +182,9 @@ module.exports = function setConvert(ax) {
             // encounters them, ie all the categories from the
             // first data set, then all the ones from the second
             // that aren't in the first etc.
-            // TODO: sorting options - do the sorting
-            // progressively here as we insert?
+            // it is assumed that this function is being invoked in the
+            // already sorted category order; otherwise there would be
+            // a disconnect between the array and the index returned
```

Comment:
@etpinard maybe an example is helpful: The first vector has X cat. values `['a', 'c', 'd']`. This _is_ already sorted. Then a subsequent series is plotted in the same chart, with values for `['a', 'b', 'c', 'd']`. By this point, the function has returned indices such as `0, 1, 2` for `['a', 'c', 'd']`, respectively. Then in the next series comes a new point `'b'` correctly inserted at index `1` but the previously returned indices will become invalid and the plot will come out nonsensical (more precisely, having checked the plot, the way it came out was consistent with this mechanics).